### PR TITLE
Update Docs: Fields “custom-widgets”

### PR DESCRIPTION
**Pull request by KancerEzeroglu**
Automatically generated by Netlify CMS

### CODE REVIEW

File: website/content/docs/custom-widgets.md [hunk 1]
```diff
@@ -1,16 +1,20 @@
 ---
 title: Creating Custom Widgets
-weight: 35
 group: fields
+weight: 35
 ---
+There are two ways to build a custom widget:
```
Comment (erezrokah) [inline website/content/docs/custom-widgets.md:6]: This seems a bit off as it comes before the introduction:
> The NetlifyCMS exposes a `window.CMS` global...

I think we can drop it and the same way we have `Writing React Components inline`, we could have another section `Writing custom widgets as a separate package`

WDTY?

File: website/content/docs/custom-widgets.md [hunk 2]
```diff
@@ -198,3 +202,92 @@ You can also return a promise from `isValid`. While the promise is pending, the
 ```
 
 **Note:** Do not create a promise inside `isValid` - `isValid` is called right before trying to persist. This means that even if a previous promise was already resolved, when the user hits 'save', `isValid` will be called again. If it returns a new promise, it will be immediately marked as "in error" until the new promise resolves.
+
+## Creating a custom widget using JSX syntax with a build step
```
Comment (erezrokah) [inline website/content/docs/custom-widgets.md:206]: We can rephrase it to `Writing custom widgets as a separate package` as mentioned before.

File: website/content/docs/custom-widgets.md [hunk 3]
```diff
@@ -198,3 +202,92 @@ You can also return a promise from `isValid`. While the promise is pending, the
...
+
+Widgets are inputs for the Netlify CMS editor interface. It's a React component that receives user input and outputs a serialized value. Those are the only rules - the component can be extremely simple, like text input, or extremely complicated, like a full-blown markdown editor. They can make calls to external services, and generally do anything that JavaScript can do.
+
+We have created a boilerplate for creating Netlify CMS widgets. You can check the [netlify-cms-widget-starter](https://github.com/netlify/netlify-cms-widget-starter) out for a starter project to fork. The starter currently provides a basic string widget, which you can customize, extend, or replace entirely with your own widget.
+
+### [](https://github.com/netlify/netlify-cms-widget-starter#getting-started)Getting started
```
Comment (erezrokah) [inline website/content/docs/custom-widgets.md:212]: We would like to drop https://github.com/netlify/netlify-cms-widget-starter.git completely.
Meaning we should detail in the docs the actions required to set up that repo.
A very slim version would be:

1. Create a directory `mkdir my-custom-widget`.
2. Navigate to the directory `cd my-custom-widget`.
3. Run `npm init`. 
4. Create a `src` directory with the files `Control.js`, `Preview.js` and `index.js`.
5. Install the following dependencies...
6. Create a Webpack configuration file...

and so on

File: website/content/docs/custom-widgets.md (hunk 1 above)
Comment (KancerEzeroglu) [reply website/content/docs/custom-widgets.md:6]: Here is what I understood, am I right?
- Remove this part completely:
Screen Shot 2020-08-13 at 15 06 26
- Change this title from "Creating a custom widget using JSX syntax with a build step" to "Writing custom widgets as a separate package".
But I think it will make the document more readable if we mention at the beginning that there are two ways to create custom widgets.
What do you think?

Comment (erezrokah) [reply website/content/docs/custom-widgets.md:6]: Yes, correct remove that section completely (like it was in the original version).

The title change is the one I'm mentioning in https://github.com/netlify/netlify-cms/pull/4144#discussion_r469923970

> But I think it will make the document more readable if we mention at the beginning that there are two ways to create custom widgets. What do you think?

Good point. How about we finish the detailed parts first and then if the doc gets too long we can section it using headers (e..g "Introduction", 'Writing Inline", "Separate Package"...)?

Comment (KancerEzeroglu) [reply website/content/docs/custom-widgets.md:6]: it makes sense
